### ISSUE TITLE
Test: MD009 rule not detecting trailing spaces in code blocks

### ISSUE THREAD

**Issue by rvben** [rvben/rumdl#203]
## Description

When running `rumdl check` on a markdown file that contains trailing spaces inside fenced code blocks, the MD009 rule does not flag them as violations.

## Steps to Reproduce

1. Create a file with trailing spaces inside a code block:
   ```markdown
   # Example
   
   ```python
   def hello():   
       print("world")  
   ```
   ```

2. Run `rumdl check test.md`
3. Expected: MD009 warnings for trailing spaces
4. Actual: No warnings shown

## Environment
- rumdl version: 0.0.165
- OS: macOS 14.0

This is a test issue to verify the n8n auto-labeling workflow.
